docs/architecture/overview.md

```diff
@@ -0,0 +1,193 @@
+
+<!--
+
+Mermaid live:
+
+https://mermaid.live/edit#pako:eNptUl1vozAQ_CvWPrUSTQkQQlBViYRWRbrecQWp0tV92IAbrAYbGXNqm-a_n4F-pOr5xfbuzHhn1zsoZMkghI3CpiJ5TAUxK-d6y-4opN16y9uKwj0VY-ZSSaGZKE3yhmGhCTbN2VqdH0UvnWIk06h5QW7ZmkRNczwQe9oSi8eRFWXp5OdFTlbSwGsueI1bEqXJgcjKPIFcMPVVY4VF1Rc17Ifw_k4epCI3rOTtJyFK-vcSkjH1lxeHlF8NE1HyiYyXBhmjxjW2h7j30KCeylZvFMt-_zj-T0PIycn561Wep6fGzOu74S_ue8hY7fdwlHyPxUsqwIKaqRp5aYa06zEUdMVqRiE0xxLVIwUq9gaHnZbZsygg1KpjFijZbSoIH3DbmlvXlKhZzNFMuv6ImoZpqa7HPzB8BQsaFBDu4AlCx55M53YwXziLwPb9meNZ8Axh4EwC33WmvuPPnbm78PcWvEhpVKcT2_UdL_C82WxqB77nDnJ_huRY1Ub1Vt4qNE6ZWslOaAhdb7H_B9Ycy5w
+
+-->
+
+:::image type="content" source="media/publish-app-topology-thumb.png" alt-text="Published app topology" lightbox="media/publish-app-topology.png":::
+
+For more information on how to publish mode, [.NET Aspire deployments](../deployment/overview.md).
+
+## Dev-time orchestration
+
+In run mode, [the app host orchestrates](../fundamentals/app-host-overview.md) all resources defined in your app model. But how does it achieve this?
+
+> [!IMPORTANT]
+> The app host isn't a production runtime. It's a development-time orchestration tool that simplifies the process of running and debugging your application locally.
+
+In this section, several key questions are answered to help you understand how the app host orchestrates your app model:
+
+- **What powers the orchestration?**
+
+  Orchestration is delegated to the [Microsoft Developer Control Plane](#developer-control-plane) (DCP), which manages resource lifecycles, startup order, dependencies, and network configurations across your app topology.
+
+- **How is the app model used?**
+
+  The app model defines all resources via implementations of <xref:Aspire.Hosting.ApplicationModel.IResource>, including containers, processes, databases, and external services—forming the blueprint for orchestration.
+
+- **What role does the app host play?**
+
+  The app host provides a high-level declaration of the desired application state. It delegates execution to DCP, which interprets the app model and performs orchestration accordingly.
+
+- **What resources are monitored?**
+
+  All declared resources—including containers, executables, and integrations—are monitored to ensure correct behavior and to support a reliable dev loop.
+
+- **How are containers and executables managed?**
+
+  Containers and processes are initialized with their configs, launched in order, and networked as needed. DCP ensures their readiness and connectivity during orchestration.
```

Comment:
"launched in order" is a bit unclear. DCP is designed to launch everything as quickly as possible, which means doing it with as much concurrency as it is allowed by the graph of dependencies (expressed via DependsOn() calls in the app model).

Reply:
`DependsOn`? Do you mean `WithReference` or `WaitFor`?

Reply:
I mean `WaitFor`